sdk/src/Services/DynamoDBv2/Custom/DataModel/Utils.cs

```diff
@@ -362,19 +359,16 @@ public static bool IsReadWrite(MemberInfo member)
         }
         public static bool ImplementsInterface(Type targetType, Type interfaceType)
         {
-            var targetTypeWrapper = TypeFactory.GetTypeInfo(targetType);
-            var interfaceTypeWrapper = TypeFactory.GetTypeInfo(interfaceType);
-            if (!interfaceTypeWrapper.IsInterface)
+            if (!interfaceType.IsInterface)
                 throw new ArgumentOutOfRangeException("interfaceType", "Type is not an interface");
 
-            foreach (var inter in targetTypeWrapper.GetInterfaces())
+            foreach (var inter in targetType.GetInterfaces())
             {
-                var interWrapper = TypeFactory.GetTypeInfo(inter);
-                if (object.Equals(interWrapper, interfaceTypeWrapper))
+                if (object.Equals(inter, interfaceType))
                     return true;
-                if (interfaceTypeWrapper.IsGenericTypeDefinition && interWrapper.IsGenericType)
+                if (targetType.IsGenericTypeDefinition && inter.IsGenericType)
```

Comment:
Shouldn't `targetType.IsGenericTypeDefinition` be `interfaceType.IsGenericTypeDefinition`?

Reply:
Great catch, fixed.